metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/AlertTaskConverter.java

```diff
@@ -0,0 +1,76 @@
+package org.hypertrace.alert.engine.metric.anomaly.task.manager.job;
+
+import static org.hypertrace.alert.engine.metric.anomaly.task.manager.job.AlertTaskJobConstants.METRIC_ANOMALY_EVENT_CONDITION;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.util.JsonFormat;
+import com.google.protobuf.util.JsonFormat.Parser;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+import java.util.Optional;
+import org.hypertrace.alert.engine.eventcondition.config.service.v1.MetricAnomalyEventCondition;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.AlertTask;
+import org.hypertrace.core.documentstore.Document;
+
+public class AlertTaskConverter {
+  private static ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final Parser JSON_PARSER = JsonFormat.parser().ignoringUnknownFields();
+
+  static final String EVENT_CONDITION_ID = "eventConditionId";
+  static final String EVENT_CONDITION_TYPE = "eventConditionType";
+  static final String EVENT_CONDITION = "eventCondition";
+  static final String DEFAULT_TENANT_ID = "__default";
+
+  public static AlertTask toAlertTask(Document document, int delayInMinutes) throws IOException {
+    JsonNode rule = OBJECT_MAPPER.readTree(document.toJson());
+
+    AlertTask.Builder builder = AlertTask.newBuilder();
+    String eventConditionType = rule.get(EVENT_CONDITION_TYPE).textValue();
+    if (eventConditionType.equals(METRIC_ANOMALY_EVENT_CONDITION)) {
+      Instant current = roundHalfDown(Instant.now(), ChronoUnit.MINUTES);
```

Comment:
we need to round it down to the last minute?

Reply:
Done. Rounding to the current minute, and based on configured delay in minutes, it will go back to the prior min.
So, if the current min is, 11s:10m:11s and `delayInMinutes` is set to 1, it will create a task with `11.08 am - 11.09 am` for reporting at 11.10am (plus some seconds).